app/uk/gov/hmrc/economiccrimelevyreturns/controllers/AmendReasonController.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyreturns.controllers
+
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.economiccrimelevyreturns.connectors.EclReturnsConnector
+import uk.gov.hmrc.economiccrimelevyreturns.controllers.actions.{AuthorisedAction, DataRetrievalAction}
+import uk.gov.hmrc.economiccrimelevyreturns.forms.AmendReasonFormProvider
+import uk.gov.hmrc.economiccrimelevyreturns.forms.FormImplicits.FormOps
+import uk.gov.hmrc.economiccrimelevyreturns.models.Mode
+import uk.gov.hmrc.economiccrimelevyreturns.navigation.AmendReasonPageNavigator
+import uk.gov.hmrc.economiccrimelevyreturns.views.html.AmendReasonView
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AmendReasonController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedAction,
+  getReturnData: DataRetrievalAction,
+  eclReturnsConnector: EclReturnsConnector,
+  formProvider: AmendReasonFormProvider,
+  pageNavigator: AmendReasonPageNavigator,
+  view: AmendReasonView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[String] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (authorise andThen getReturnData) { implicit request =>
+    Ok(view(form.prepare(request.eclReturn.amendReason), mode, request.startAmendUrl))
```

Comment:
`AmendReasonController` will always be called within amend journey, so there is no need to pass `startAmendUrl` as parameter. instead of this you can remove it from here, from view parameter list and set start amend url to start of amend journey

Reply:
Thus URL is constructed by the previous controller with parameters that only it has. I prefver passing it to the view than calling the back-end again to reconstruct it in the view.

Reply:
Ah, I see, it has other variables there as well.